### PR TITLE
[SQIP] Add documentation

### PR DESCRIPTION
**Pull request by axe312ger**
My working PR for the upcoming SQIP documentation. This is not ready yet, I just prefer pull requests to track my own progress.

Comments are still welcome, ofc :)

Readme preview: https://github.com/axe312ger/gatsby/tree/docs/using-sqip/packages/gatsby-transformer-sqip
Example preview: https://deploy-preview-5287--using-sqip.netlify.com/

### CODE REVIEW

File: examples/using-sqip/src/components/polaroid.js [hunk 1]
```diff
@@ -48,6 +48,8 @@ const data = {
   },
 }
 
+// @todo move all the styles to css file
```
Comment (axe312ger) [inline examples/using-sqip/src/components/polaroid.js]: @todo

File: examples/using-sqip/src/pages/index.js [hunk 2]
```diff
@@ -3,7 +3,21 @@ import React from 'react'
...
+    <p>It is a svg based implementation of low quality image previews (LQIP)</p>
+    <p>
+      <strong>More precisely:</strong>
+      <br /> An algorithm calculates a primitive representation of your images
+      based on simple shapes like circles, ellipses, triangles and more. These
+      will be embedded in your initial HTML payload. This will help your users
+      to get a feeling of how the pictures will look like, even{' '}
+      <strong>before</strong> they got loaded by their (probably) slow
+      connection.
+    </p>
```
Comment (axe312ger) [inline examples/using-sqip/src/pages/index.js]: non-jsx example missing here.

Add copyright hint for background image, too

File: packages/gatsby-transformer-sqip/README.md [hunk 3]
```diff
@@ -2,25 +2,144 @@
 
 Generates vectorized primitive version of images to be used as preview thumbnails.
 
+
+@todo add some example gif
```
Comment (axe312ger) [inline packages/gatsby-transformer-sqip/README.md]: @todo add example gif

File: packages/gatsby-transformer-sqip/README.md [hunk 4]
```diff
@@ -30,4 +149,31 @@ const Img = require(`gatsby-image`)
...
+* Images fade in all the time, even when the image is already in the browser cache
+
+
+## :books: Settings recommendations
+
+The plugin offers a wide variety of settings, which affect the size of the resulting preview image.
+
+Some research: https://axe312ger.github.io/embedded-svg-loading-impact-research/
+
+@todo image with loading time impact from webpagetest.com
```
Comment (axe312ger) [inline packages/gatsby-transformer-sqip/README.md]: @todo this whole section is still very incomplete

**Comment by m-allanson** [comment]
This looks like it's coming together nicely, it's great to see detailed docs like this :) Drop a note here whenever this is ready for review.

**Comment by axe312ger** [comment]
Especially @m-allanson but for everybody: This is ready for review.

The only missing piece is the example gif for the readme, will follow soom :tm:

**Comment by KyleAMathews** [comment]
Deploy preview for *using-glamor* failed.

Built with commit 52cfd5cffab5609fa822c97533b52a1108d9983a

https://app.netlify.com/sites/using-glamor/deploys/5b0ff73011b73b2d6d8fa4f8

**Comment by axe312ger** [comment]
@KyleAMathews 

> 3:27:36 PM: error Couldn't find any versions for "gatsby-cli" that matches "^1.1.54"

Same as my local issue as mentioned on discord 🙈

**Comment by m-allanson** [comment]
Looking 👍!

Is there an invisible character in the _Using_ section? I'm seeing a question mark:

screen shot 2018-05-31 at 21 03 45

Otherwise I think this is ready to go in once the gif is added.

File: packages/gatsby-transformer-sqip/README.md [hunk 5]
```diff
@@ -2,25 +2,149 @@
...
+image {
+  sqip(numberOfPrimitives: 12, blur: 12, width: 256, height: 256),
+  sizes(maxWidth: 400, maxHeight: 400) {
+    ...GatsbyImageSharpSizes_noBase64
+  }
+}
+```
+
+With `
gatsby-transformer-contentful`:
```
Comment (pieh) [inline packages/gatsby-transformer-sqip/README.md]: This should be with `gatsby-source-contentful` ? :)

**Comment by axe312ger** [comment]
I guess this is ready for merge or at least some re-reviews :)

@pieh @m-allanson @KyleAMathews

**Comment by axe312ger** [comment]
If ✅ is given, this should be squash-merged into one commit :)

**Pull request merged**